third-party/cmake/BuildLuajit.cmake

```diff
@@ -1,25 +1,81 @@
-ExternalProject_Add(luajit
-  PREFIX ${DEPS_BUILD_DIR}
-  URL ${LUAJIT_URL}
-  DOWNLOAD_DIR ${DEPS_DOWNLOAD_DIR}/luajit
-  DOWNLOAD_COMMAND ${CMAKE_COMMAND}
-    -DPREFIX=${DEPS_BUILD_DIR}
-    -DDOWNLOAD_DIR=${DEPS_DOWNLOAD_DIR}/luajit
-    -DURL=${LUAJIT_URL}
-    -DEXPECTED_SHA256=${LUAJIT_SHA256}
-    -DTARGET=luajit
-    -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/DownloadAndExtractFile.cmake
-  CONFIGURE_COMMAND ""
-  BUILD_IN_SOURCE 1
-  BUILD_COMMAND ""
-  INSTALL_COMMAND ${MAKE_PRG} CC=${DEPS_C_COMPILER}
-                              PREFIX=${DEPS_INSTALL_DIR}
-                              CFLAGS=-fPIC
-                              CFLAGS+=-DLUAJIT_DISABLE_JIT
-                              CFLAGS+=-DLUA_USE_APICHECK
-                              CFLAGS+=-DLUA_USE_ASSERT
-                              CCDEBUG+=-g
-                              BUILDMODE=static
-                              install)
+# Reusable function to build Luajit, set the following build commands before
+# calling it:
+#   LUAJIT_CONFIGURE_COMMAND
+#   LUAJIT_BUILD_COMMAND
+#   LUAJIT_INSTALL_COMMAND
```

Comment:
We could use `cmake_parse_arguments()` here and pass arguments to the function too (like we do in `InstallHelpers.cmake`).  I think we should probably do that some other time, and keep this focused on getting something working in place.